app/helpers/comment_helper.rb

```diff
@@ -96,18 +107,150 @@ def comment_destroy_helper(comment)
     if comment.destroy
       case comment.type
       when 'StepComment'
-        log_activity(:delete_step_comment)
+        log_step_activity(:delete_step_comment, comment)
       when 'ResultComment'
-        log_activity(:delete_result_comment)
+        log_result_activity(:delete_result_comment, comment)
       when 'ProjectComment'
-        log_activity(:delete_project_comment)
+        log_project_activity(:delete_project_comment, comment)
       when 'TaskComment'
-        log_activity(:delete_module_comment)
+        log_my_module_activity(:delete_module_comment, comment)
       end
       render json: {}, status: :ok
     else
       render json: { message: I18n.t('comments.delete_error') },
                  status: :unprocessable_entity
     end
   end
+
+  def result_comment_annotation_notification(comment, old_text = nil)
```

Comment:
Hmm... 
Can we abstract all these notifications in one CommentNotificationService? 
Or/And implement the `send_notification` method on each Comment SubClass? 

Also, those send notifications can be executed asynchronous...